src/components/ActionsMenuLayout/ActionsMenuLayout.tsx

```diff
@@ -87,42 +89,35 @@ export class ActionsMenuLayout extends React.Component<ActionsMenuLayoutProps> {
     }
   };
 
-  _onFocus = () => {
-    const { focusedIdx } = this.state;
-    const { focusedIndex } = this.props;
-
-    this._focusItem(
-      typeof focusedIndex !== 'undefined' ? focusedIndex : focusedIdx,
+  _onFocus = (e) => {
+    const focusOnChild = !!Object.values(this.liRefs).find(
+      (i) => i.liRef?.current === e.target,
     );
-  };
-
-  // to get proper index on mouse click
-  _onClick = (e) => {
-    const indexValue = Object.values(this.liRefs).findIndex((i) => {
-      return i.liRef?.current === e.target;
-    });
-
-    if (indexValue >= 0) {
-      this._focusItem(indexValue);
+    //focus on initial list item after tab focusing
+    if (!focusOnChild) {
+      this._focusItem(0);
     }
   };
 
+  //prevent firing focus after mouse click events
+  _preventFocus = (e: React.SyntheticEvent) => e.preventDefault();
+
   render() {
     const { alignment, children, className } = this.props;
     return (
       <TPAComponentsConsumer>
         {({ mobile }) => (
           <ul
             onFocus={this._onFocus}
-            onClick={this._onClick}
+            onMouseDown={this._preventFocus}
```

Comment:
Is this still needed here?

Reply:
@jonathanadler  depends, these lines you commented on are related. 
so now you can tab-focus the actions menu component and the first item in the list will focused immediately. to achieve this needed to make container tab-able (with tabIndex), but then we need to prevent the focus event from firing, as they will be fired when clicking on list item or the action menu container.

now, if the way forward is to have `ActionMenu` component which will control the focus, ability to tab into this component is not actually needed. what do you think?

Reply:
oh, I see cool 👍